src/lib/Types/Primitives.hs

```diff
@@ -139,10 +141,7 @@ data PrimOp e =
       | SumToVariant e
       -- Pointer to the stdout-like output stream
       | OutputStreamPtr
-      | SynthesizeDict SrcPosCtx e  -- Only used during type inference
-      | ProjMethod e e  -- project a method, by source name, from the dict
-      -- Used in prelude for `run_accum`. Only works for single-method classes.
-      | ExplicitDict e e  -- dict type, dict method
+      | ProjMethod e Int  -- project a method from the dict
```

Comment:
Because method projection is now where work can happen! Previously, synthesizing a dictionary for `Add (n=>a)` would create an array of zeros for the `zero` method. Now, that only happens at method projection, when you actually projection out the `zero` method. This means we don't have to play tricks with thunking either at the user level or inside the compiler.